packages/flutter/lib/src/widgets/framework.dart

```diff
@@ -3070,12 +3070,33 @@ abstract class Element extends DiagnosticableTree implements BuildContext {
       return null;
     }
     if (child != null) {
-      if (child.widget == newWidget) {
+      bool canUpdate = true;
+      // When the type of a widget is changed between Stateful and Stateless via
+      // hot reload, the element tree will end up in a partially invalid state.
+      // That is, if the widget was a StatefulWidget and is now a StatelessWidget,
```

Comment:
Does that mean when a class change its superclass, the runtimeType is still the same? That feels like a compiler bug?

Reply:
A class can only change its type like this via hot reload. It isn't that the runtimeType is still the same, its that they're both different

Reply:
... differently the same

Reply:
I see, thanks for explaining.